vulnerabilities/importers/osv.py

```diff
@@ -53,18 +65,23 @@ def parse_advisory_data(raw_data: dict, supported_ecosystem) -> Optional[Advisor
 
     for affected_pkg in raw_data.get("affected") or []:
         purl = get_affected_purl(affected_pkg=affected_pkg, raw_id=raw_id)
-        if purl.type != supported_ecosystem:
+        if purl.type in PURL_TYPE_BY_OSV_SCHEME.keys():
```

Comment:
```suggestion
        if purl.type in PURL_TYPE_BY_OSV_SCHEME:
```
.... but I am not sure I get what is happening there.... could you elaborate with an example? Why do you replace the purl type after the fact? Why not create a proper purl correctly in the first place?

Reply:
I thought to would be easier to just replace the purl type instead of create a new one from scratch 
for example the osv purl is [`pkg:rubygems/sprout`](https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2017/10/GHSA-229r-pqp6-8w6g/GHSA-229r-pqp6-8w6g.json#L17) to `pkg:gem/sprout`